scrape/manager.go

```diff
@@ -40,18 +40,23 @@ func NewManager(logger log.Logger, app Appendable) *Manager {
 		scrapeConfigs: make(map[string]*config.ScrapeConfig),
 		scrapePools:   make(map[string]*scrapePool),
 		graceShut:     make(chan struct{}),
+		targetsAll:    make(map[string][]*Target),
 	}
 }
 
 // Manager maintains a set of scrape pools and manages start/stop cycles
 // when receiving new target groups form the discovery manager.
 type Manager struct {
-	logger        log.Logger
-	append        Appendable
-	scrapeConfigs map[string]*config.ScrapeConfig
-	scrapePools   map[string]*scrapePool
-	mtx           sync.RWMutex
-	graceShut     chan struct{}
+	logger         log.Logger
+	append         Appendable
+	scrapeConfigs  map[string]*config.ScrapeConfig // Guarded by `mtx`
+	scrapePools    map[string]*scrapePool          // Guarded by `mtx`
+	targetsActive  []*Target                       // Guarded by `mtxTargets`
+	targetsDropped []*Target                       // Guarded by `mtxTargets`
+	targetsAll     map[string][]*Target            // Guarded by `mtxTargets`
+	graceShut      chan struct{}
+	mtx            sync.Mutex
+	mtxTargets     sync.Mutex
```

Comment:
Would it make sense to call `mtx` `mtxScrape` instead? I also generally like grouping a mutex with the fields that it protects, like:

```go
type Manager struct {
	logger         log.Logger
	append         Appendable

	mtxScrape      sync.Mutex // Guards the fields below.
	scrapeConfigs  map[string]*config.ScrapeConfig
	scrapePools    map[string]*scrapePool

	mtxTargets     sync.Mutex // Guards the fields below.
	targetsActive  []*Target
	targetsDropped []*Target
	targetsAll     map[string][]*Target

	graceShut      chan struct{}
}
```

Reply:
I just now saw the discussion about this on an earlier revision, and @brian-brazil suggested the grouping in https://github.com/prometheus/prometheus/pull/4005#discussion_r178528210 too. I think I still like the grouping best.

Reply:
yep , easyer to understand so far. Updated. Thanks